controlplane/api/v1beta1/conversion.go

```diff
@@ -72,25 +73,41 @@ func Convert_v1beta2_KThreesConfigSpec_To_v1beta1_KThreesConfigSpec(in *bootstra
 
 // ConvertTo converts the v1beta1 KThreesControlPlane receiver to a v1beta2 KThreesControlPlane.
 func (in *KThreesControlPlane) ConvertTo(dstRaw ctrlconversion.Hub) error {
-	dst := dstRaw.(*cabp3v1.KThreesControlPlane)
+	dst := dstRaw.(*controlplanev1beta2.KThreesControlPlane)
 	if err := Convert_v1beta1_KThreesControlPlane_To_v1beta2_KThreesControlPlane(in, dst, nil); err != nil {
 		return fmt.Errorf("converting KThreesControlPlane v1beta1 to v1beta2: %w", err)
 	}
+
+	restored := &controlplanev1beta2.KThreesControlPlane{}
+	if ok, err := utilconversion.UnmarshalData(in, restored); err != nil {
+		return fmt.Errorf("unmarshalling stored conversion data: %w", err)
+	} else if !ok {
+		// No stored data.
+		return nil
+	}
+
+	dst.Spec.KThreesConfigSpec.ServerConfig.CloudProviderName = restored.Spec.KThreesConfigSpec.ServerConfig.CloudProviderName
+	dst.Spec.KThreesConfigSpec.ServerConfig.DeprecatedDisableExternalCloudProvider = restored.Spec.KThreesConfigSpec.ServerConfig.DeprecatedDisableExternalCloudProvider
+	dst.Spec.KThreesConfigSpec.ServerConfig.DisableCloudController = restored.Spec.KThreesConfigSpec.ServerConfig.DisableCloudController
 	return nil
 }
 
 // ConvertFrom converts the v1beta1 KThreesControlPlane receiver from a v1beta2 KThreesControlPlane.
 func (in *KThreesControlPlane) ConvertFrom(srcRaw ctrlconversion.Hub) error {
-	src := srcRaw.(*cabp3v1.KThreesControlPlane)
+	src := srcRaw.(*controlplanev1beta2.KThreesControlPlane)
 	if err := Convert_v1beta2_KThreesControlPlane_To_v1beta1_KThreesControlPlane(src, in, nil); err != nil {
```

Comment:
Do we need update Convert_v1beta2_KThreesControlPlane_To_v1beta1_KThreesControlPlane as well?

Reply:
This is not needed. The generated conversion code is granular enough that will eventually call the custom `Convert_v1beta1_KThreesServerConfig_To_v1beta2_KThreesServerConfig` from the bootstrap package.